parsers/ELEXON.py

```diff
@@ -124,337 +223,137 @@ def query_additional_eso_data(
-
-    from_date = (target_datetime - timedelta(days=1)).strftime("%Y-%m-%d")
-    to_date = target_datetime.strftime("%Y-%m-%d")
-
-    params = {"FromDate": from_date, "ToDate": to_date, "ServiceType": "csv"}
-    response = query_ELEXON("INTERFUELHH", session, params)
-    return response.text
-
-
-def query_production(
-    session: Session, target_datetime: datetime | None = None, report: str = "B1620"
-):
-    if target_datetime is None:
-        target_datetime = datetime.now()
-
-    # we can only fetch one date at a time.
-    # if target_datetime is first 30 minutes of the day fetch the day before.
-    # otherwise fetch the day of target_datetime.
-    if target_datetime.time() <= time(0, 30):
-        settlement_date = target_datetime.date() - timedelta(1)
-    else:
-        settlement_date = target_datetime.date()
-
+    eso_data = response.json()["result"]["records"]
+    return eso_data
+
+
+def parse_eso_production(
+    production_data: list[dict[str, any]], logger: Logger
+) -> ProductionBreakdownList:
+    all_production_breakdowns: list[ProductionBreakdownList] = []
+    for event in production_data:
+        production_breakdown = ProductionBreakdownList(logger=logger)
+        event_datetime = parse_datetime(
+            event.get("SETTLEMENT_DATE"), event.get("SETTLEMENT_PERIOD")
+        )
+        production_mix = ProductionMix()
+        storage_mix = StorageMix()
+        for production_mode in ESO_FUEL_MAPPING.keys():
+            if ESO_FUEL_MAPPING[production_mode] == "hydro storage":
+                storage_mix.add_value("hydro", event.get(production_mode))
+                production_breakdown.append(
+                    zoneKey=ZoneKey("GB"),
+                    storage=storage_mix,
+                    source=ESO_SOURCE,
+                    datetime=event_datetime,
+                )
+            else:
+                production_mix.add_value(
+                    ESO_FUEL_MAPPING[production_mode], event.get(production_mode)
+                )
+                production_breakdown.append(
+                    zoneKey=ZoneKey("GB"),
+                    production=production_mix,
+                    source=ESO_SOURCE,
+                    datetime=event_datetime,
+                )
+
+        all_production_breakdowns.append(production_breakdown)
+    events = ProductionBreakdownList.merge_production_breakdowns(
+        all_production_breakdowns, logger
+    )
+    return events
+
+def parse_eso_hydro_storage(eso_data: list[dict[str, any]], logger: Logger ):
+    """Parses only hydro storage data from the ESO API. This data will be merged with the B1620 data"""
+    storage_breakdown = ProductionBreakdownList(logger=logger)
+    for event in eso_data:
+        event_datetime = parse_datetime(
+            event.get("SETTLEMENT_DATE"), event.get("SETTLEMENT_PERIOD")
+        )
+        storage_mix = StorageMix()
+
+        storage_mix.add_value("hydro", float(event.get("PUMP_STORAGE_PUMPING")))
```

Comment:
```suggestion
        storage_mix.add_value("hydro", float(event.get("PUMP_STORAGE_PUMPING")) if event.get("PUMP_STORAGE_PUMPING") else None)
```
Might make sense to extract it to a temp variable to avoid calling get twice but it's not needed.